### PR TITLE
Added migration tables for the application

### PR DESCRIPTION
**Pull request by UzunIon**

### CODE REVIEW

File: database/migrations/2021_12_09_171115_create_locations_table.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('locations', function (Blueprint $table) {
+            $table->id();
+            $table->string('name', 30);
+            $table->text('description');
+            $table->integer('price');
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171115_create_locations_table.php:20]: Are you sure that you always will have integer numbers?

File: database/migrations/2021_12_09_171850_create_activities_table.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    public function up()
+    {
+        Schema::create('activities', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('restaurant_id')
+            ->constrained()
+            ->cascadeOnUpdate()
+            ->restrictOnDelete();
+            $table->string('activity_type', 30);
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171850_create_activities_table.php]: can't you use just `type` for column name?

File: database/migrations/2021_12_09_171903_create_tours_table.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    public function up()
+    {
+        Schema::create('tours', function (Blueprint $table) {
+            $table->id();
+            
+            $table->string('name', 50);
+            $table->text('details');
+
+            $table->foreignId('guest_house_id')
+            ->constrained()
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171903_create_tours_table.php]: please fix indentation.

File: database/migrations/2021_12_09_171903_create_tours_table.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    public function up()
+    {
+        Schema::create('tours', function (Blueprint $table) {
+            $table->id();
+            
+            $table->string('name', 50);
+            $table->text('details');
+
+            $table->foreignId('guest_house_id')
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171903_create_tours_table.php]: I recommend to keep foreign columns at top near primary key

File: database/migrations/2021_12_09_172010_create_contact_info_table.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                PRIMARY KEY (`customer_id`),
+                UNIQUE KEY (`customer_id`),
+                CONSTRAINT `fk_contact_info_customers`
+                FOREIGN KEY (`customer_id`) REFERENCES `customers` (`id`)
+                ON DELETE CASCADE
+                ON UPDATE CASCADE
+            ) ENGINE=InnoDB;
+        ");
+
+        // Schema::create('contact_info', function (Blueprint $table) {
```
Comment (egorbwork) [inline database/migrations/2021_12_09_172010_create_contact_info_table.php]: commented code won't be accepted.

File: database/migrations/2021_12_11_105623_alter_contact_info_table_add_gender_column.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('contact_info');
```
Comment (egorbwork) [inline database/migrations/2021_12_11_105623_alter_contact_info_table_add_gender_column.php]: invalid rollback

File: database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('activities');
```
Comment (egorbwork) [inline database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php]: invalid rollback

**Comment by egorbwork** [comment]
Alter changing column is missing.

**Review by egorbwork — CHANGES_REQUESTED** [review]

File: database/migrations/2021_12_09_171115_create_locations_table.php (hunk 1 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171115_create_locations_table.php:20]: No, modified data type in decimal

File: database/migrations/2021_12_09_171850_create_activities_table.php (hunk 2 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171850_create_activities_table.php]: Changed column activity_type name to type

File: database/migrations/2021_12_09_171903_create_tours_table.php (hunk 3 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171903_create_tours_table.php]: Fixed

File: database/migrations/2021_12_09_171903_create_tours_table.php (hunk 4 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171903_create_tours_table.php]: Put all foreign columns after the primary keys

File: database/migrations/2021_12_09_172010_create_contact_info_table.php (hunk 5 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_172010_create_contact_info_table.php]: Deleted commented code

File: database/migrations/2021_12_11_105623_alter_contact_info_table_add_gender_column.php (hunk 6 above)
Comment (UzunIon) [reply database/migrations/2021_12_11_105623_alter_contact_info_table_add_gender_column.php]: Fixed

File: database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php (hunk 7 above)
Comment (UzunIon) [reply database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php]: Fixed

File: database/migrations/2021_12_09_170805_create_customers_table.php [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('customers', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
```
Comment (egorbwork) [inline database/migrations/2021_12_09_170805_create_customers_table.php]: length is missing.for both varchar columns.

File: database/migrations/2021_12_09_170805_create_customers_table.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            $table->string('surname');
+            $table->timestamp('created_at')->useCurrent();
```
Comment (egorbwork) [inline database/migrations/2021_12_09_170805_create_customers_table.php]: why not datetime?

File: database/migrations/2021_12_09_171115_create_locations_table.php (hunk 1 above)
Comment (egorbwork) [reply database/migrations/2021_12_09_171115_create_locations_table.php:20]: no changes here.

File: database/migrations/2021_12_09_171850_create_activities_table.php [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('activities', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('restaurant_id')
+            ->constrained()
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171850_create_activities_table.php]: Please fix indentation.

File: database/migrations/2021_12_09_171903_create_tours_table.php (hunk 3 above)
Comment (egorbwork) [reply database/migrations/2021_12_09_171903_create_tours_table.php]: should be 4 spaces on right.

File: database/migrations/2021_12_09_171903_create_tours_table.php [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+            $table->id();
+            $table->foreignId('guest_house_id')
+            ->constrained()
+            ->cascadeOnUpdate()
+            ->restrictOnDelete();
+            $table->foreignId('location_id')
+            ->constrained()
+            ->cascadeOnUpdate()
+            ->restrictOnDelete();
+            $table->foreignId('sport_activity_id')
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171903_create_tours_table.php]: maybe we should accept it as nullable?

File: database/migrations/2021_12_09_171935_create_orders_table.php [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('orders', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('customer_id')
+            ->constrained()
+            ->restrictOnUpdate()
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171935_create_orders_table.php]: I recommend cascade on update.

File: database/migrations/2021_12_09_171935_create_orders_table.php [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+            ->cascadeOnDelete();
+            $table->foreignId('tour_id')
+            ->constrained()
+            ->restrictOnUpdate()
+            ->cascadeOnDelete();
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171935_create_orders_table.php]: looks as here cascade should be on update and restrict on delete

File: database/migrations/2021_12_09_172010_create_contact_info_table.php [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    public function up()
+    {
+        DB::statement("
+            CREATE TABLE contact_info (
+                `customer_id` BIGINT UNSIGNED NOT NULL,
+                `phone_number` VARCHAR(20) NOT NULL,
+                `email` VARCHAR(50) NOT NULL,
+                PRIMARY KEY (`customer_id`),
+                UNIQUE KEY (`customer_id`),
```
Comment (egorbwork) [inline database/migrations/2021_12_09_172010_create_contact_info_table.php]: You don't need unique key if you already have Primary key for it.

File: database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('locations', function (Blueprint $table) {
+            $table->decimal('price', 5,2);
```
Comment (egorbwork) [inline database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php]: should be change in this case.

File: database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('locations', function (Blueprint $table) {
+            $table->integer('price');
```
Comment (egorbwork) [inline database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php]: should be change in this case.

File: database/migrations/2021_12_09_171115_create_locations_table.php (hunk 1 above)
Comment (egorbwork) [reply database/migrations/2021_12_09_171115_create_locations_table.php:20]: But it is migration to fix it, it is acceptable. But migration for change should be fixed.

**Review by egorbwork — CHANGES_REQUESTED** [review]

**Comment by egorbwork** [comment]
Better, but some moments require fixes.

File: database/migrations/2021_12_09_170805_create_customers_table.php (hunk 8 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_170805_create_customers_table.php]: Fixed

File: database/migrations/2021_12_09_170805_create_customers_table.php (hunk 9 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_170805_create_customers_table.php]: Read about the differences, modified to datetime

File: database/migrations/2021_12_09_171850_create_activities_table.php (hunk 10 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171850_create_activities_table.php]: This time fixed for sure

File: database/migrations/2021_12_09_171935_create_orders_table.php (hunk 12 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171935_create_orders_table.php]: Fixed

File: database/migrations/2021_12_09_171935_create_orders_table.php (hunk 13 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171935_create_orders_table.php]: Fixed

File: database/migrations/2021_12_09_172010_create_contact_info_table.php (hunk 14 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_172010_create_contact_info_table.php]: Deleted unique key

File: database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php (hunk 15 above)
Comment (UzunIon) [reply database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php]: Fixed

File: database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php (hunk 16 above)
Comment (UzunIon) [reply database/migrations/2021_12_13_170750_alter_locations_table_change_price_data_type.php]: Fixed

File: database/migrations/2021_12_09_171903_create_tours_table.php (hunk 11 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171903_create_tours_table.php]: If I understand correctly, this solves the problem of the obligation to attend a sports activity on the tour, added nullable

File: database/migrations/2021_12_09_171715_create_sport_activities_table.php [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('sport_activities', function (Blueprint $table) {
+            $table->id();
+            $table->string('name', 20);
+            $table->text('description');
+            $table->dateTime('active_from');
```
Comment (egorbwork) [inline database/migrations/2021_12_09_171715_create_sport_activities_table.php]: Initialy you planned to use date. DateTime will complicate your life fir this columns.

File: database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        //
```
Comment (egorbwork) [inline database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php]: Rollback is missing.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Missing rollback and changes from different homework is not looking good.

File: database/migrations/2021_12_09_171715_create_sport_activities_table.php (hunk 17 above)
Comment (UzunIon) [reply database/migrations/2021_12_09_171715_create_sport_activities_table.php]: I get it, fixed

File: database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php (hunk 18 above)
Comment (UzunIon) [reply database/migrations/2021_12_11_113122_alter_activities_table_drop_duration_column.php]: Fixed the mistake

**Review by egorbwork — APPROVED** [review]
Changes for project models now are acceptable. The blog ones should be reviewed by Victor. So it has combination of 2 homeworks, that is complicating review. Also I see removal of files, I hope it was done with keeping them physically in other way the application after merging this pr, will be dead.

**Pull request merged**